### PR TITLE
improved display format for Z syntax errors

### PR DESCRIPTION
**Pull request by mccanne**

### CODE REVIEW

File: zql/parser-support.go [hunk 1]
```diff
@@ -114,3 +116,56 @@ func makeUnicodeChar(chars interface{}) string {
 
 	return string(r)
 }
+
+var syntaxRegexp = regexp.MustCompile(`^([0-9]+):([0-9]+) \(([0-9]+)\): no match found`)
+
+func ImproveError(src string, e error) error {
+	fmt.Printf("===\n%s===\n", src)
```
Comment (nwt) [inline zql/parser-support.go:123]:
```suggestion
```

File: zql/parser-support.go [hunk 2]
```diff
@@ -114,3 +116,56 @@ func makeUnicodeChar(chars interface{}) string {
...
+		return e
+	}
+	if colNo > 100 {
+		return e
+	}
+	_, err = strconv.Atoi(hits[3])
+	if err != nil {
+		return e
+	}
+	lineNo--
```
Comment (nwt) [inline zql/parser-support.go:143]:
```suggestion
	el, ok := e.(errList)
	if !ok || len(el) != 1 {
		return e
	}
	pe, ok := el[0].(*parserError)
	if !ok {
		return e
	}
	lineNo, colNo := pe.pos.line-1, pe.pos.col
```

File: zql/parser-support.go [hunk 3]
```diff
@@ -114,3 +116,56 @@ func makeUnicodeChar(chars interface{}) string {
 
 	return string(r)
 }
+
+var syntaxRegexp = regexp.MustCompile(`^([0-9]+):([0-9]+) \(([0-9]+)\): no match found`)
+
+func ImproveError(src string, e error) error {
```
[the remark this answers is not in the document]
Comment (mccanne) [reply zql/parser-support.go:122]: Good idea but that method is generated by pigeon so I will use `ParseZ`.

File: zql/parser-support.go (hunk 1 above)
Comment (mccanne) [reply zql/parser-support.go:123]: oops

File: zql/parser-support.go (hunk 2 above)
Comment (mccanne) [reply zql/parser-support.go:143]: doh!  I went to do that and saw the error values weren't exported without thinking this code lives in the same package :-)

File: compiler/compiler.go [hunk 4]
```diff
@@ -13,16 +13,16 @@ import (
 
 // ParseProc() is an entry point for use from external go code,
 // mostly just a wrapper around Parse() that casts the return value.
-func ParseProc(query string, opts ...zql.Option) (ast.Proc, error) {
-	parsed, err := zql.Parse("", []byte(query), opts...)
+func ParseProc(z string) (ast.Proc, error) {
+	parsed, err := zql.ParseZ(z)
 	if err != nil {
-		return nil, err
+		return nil, zql.ImproveError(z, err)
```
Comment (nwt) [inline compiler/compiler.go:19]: Don't want this with zql.ParseZ.
```suggestion
		return nil, err
```

**Review by nwt — APPROVED** [review]

File: compiler/compiler.go (hunk 4 above)
Comment (mccanne) [reply compiler/compiler.go:19]: I'm getting old, haha

**Pull request merged**